reader-writer-lock/src/main/java/com/iluwatar/reader/writer/lock/ReaderWriterLock.java

```diff
@@ -0,0 +1,147 @@
+package com.iluwatar.reader.writer.lock;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class responsible for control the access for reader or writer
+ */
+public class ReaderWriterLock {
+
+  /**
+   * Mutex for reader
+   */
+  private Object r = new Object();
```

Comment:
single character names should not be promoted, I would propose `mutex` or `readMutex` something in that line which also describes what it is about. And then the javadoc on private variable will not be needed.